src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityBarbarian.java

```diff
@@ -0,0 +1,159 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import com.minecolonies.coremod.sounds.BarbarianSounds;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.*;
+import net.minecraft.entity.monster.EntityMob;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class describing the Barbarian Entity
+ */
+public class EntityBarbarian extends EntityMob
+{
+    private final Colony colony = ColonyManager.getClosestColony(world, this.getPosition());
+
+    /**
+     * defines the default values for the Entity's attributes.
+     */
+    private static final double FOLLOW_RANGE          = 35.0D;
+    private static final double MOVEMENT_SPEED        = 0.2D;
+    private static final double ATTACK_DAMAGE         = 2.0D;
+    private static final double ARMOR                 = 2.0D;
+    private static final double BARBARIAN_BASE_HEALTH = 25;
+
+    /**
+     * Defines the default values for the various AI Task's priorities.
+     */
+    private static final int   PRIORITY_ZERO      = 1;
+    private static final int   PRIORITY_TWO       = 2;
+    private static final int   PRIORITY_THREE     = 3;
+    private static final int   PRIORITY_FOUR      = 4;
+    private static final int   PRIORITY_FIVE      = 5;
+    private static final int   PRIORITY_SIX       = 6;
+    private static final int   PRIORITY_EIGHT     = 8;
+    private static final float MAX_WATCH_DISTANCE = 8.0F;
+
+    public static final ResourceLocation LOOT = new ResourceLocation(Constants.MOD_ID, "EntityBarbarianDrops");
+
+    /**
+     * Constructor method for entity
+     *
+     * @param worldIn The world that the entity is in
+     */
+    public EntityBarbarian(final World worldIn)
+    {
+        super(worldIn);
+    }
+
+    /**
+     * Applies the following attributes to the entity
+     */
+    @Override
+    protected void applyEntityAttributes()
```

Comment:
same here. the problem with javadoc on override is that if we change the overrided method, the javadoc will not be reflected here